### PR TITLE
Backspace on empty query in FilterBox navigates back

### PR DESCRIPTION
**Pull request by joadoumie**
This is a feature request @plante-msft has had for a while, but I don't see an open issue for it.

Does what title says, navigates backwards when a user types `Backspace` and the filterbox is already empty.

### CODE REVIEW

**Comment by zadjii** [comment]
I suspect we might be able to use a `PreviewKeyDown` handler on the search box instead. That would prevent the need to store the previous search text, since `PreviewKeyDown` would fire _before_ the text changed (so you could know when backspace was pressed for an already empty search)

**Comment by joadoumie** [comment]
> I suspect we might be able to use a `PreviewKeyDown` handler on the search box instead. That would prevent the need to store the previous search text, since `PreviewKeyDown` would fire _before_ the text changed (so you could know when backspace was pressed for an already empty search)

I didn't even know that was a thing. That seems way better. Lemme try that.

**Comment by joadoumie** [comment]
Changed to a draft because: 

1. Definitely the preview handler feels like the right way to go but....
2. Everytime the palette loses focus and then regains focus it stops being triggered and I don't have time to finish debugging atm. Will convert back to a normal PR and request review once the time is right.

**Comment by joadoumie** [comment]
@zadjii-msft - I made some changes so the functionality works as follows: 

1. If a user presses down on the backspace key with an empty query and holds the backspace key, cmdpal will not navigate back until the key is released 

2. If a user presses down on the backspace key with a non-empty query and holds the backspace key, cmdpal will not navigate back even after the key is released. This is to ensure that folks can hold backspace and delete the entire text field without inadvertently getting sent backwards in the flow. In this case, they would have to explicitly hit backspace again. 

TLDR - hitting backspace will only ever navigate back on release AND if backspace was originally hit with an empty query.

**Review by zadjii-msft — APPROVED** [review]
